pycqed/measurement/waveform_control/segment.py

```diff
@@ -1070,7 +1081,7 @@ def plot(self, instruments=None, channels=None, legend=True,
                     a.legend(loc=[1.02, 0], prop={'size': 8})
                 a.set_ylabel('Voltage (V)')
             ax[-1, 0].set_xlabel('time ($\mu$s)')
-            fig.suptitle(f'{self.name}')
+            fig.suptitle(f'{self.name}', y=1.01)
             plt.tight_layout()
             if savefig:
                 plt.savefig(f'{self.name}.png')
```

Comment:
I think at some point I added a plt.close(fig). Is this not needed?

Reply:
I guess this was removed when the function was changed by Nathan to not show the plots, but return them instead. If we close them here, they cannot be shown afterwards. Since they are returned, the caller is responsible for closing them (which we probably don't do, when calling it from a notebook.) Not sure, what behavior we want. Producing screen output inside a function is not desirable either, that is why Nathan remove the .show(). However, we could add parameters closefig and plotfig, and issue a warning if closefig and not savefig and not plotfig. What do you think?

Reply:
Ah, no, it was apparently not removed: you have changed it on the parallel 2qb calib branch, that is why it is not in the master. But then you return a closed figure, which does not really help.

Reply:
I think we should have the flag close_fig=True by default. And if close_fig then it calls plt.show() and does not return figure, else it does not call plt.show() + returns figure.